web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -0,0 +1,414 @@
+    !selectedCluster ||
+    enrollmentState.status !== 'notStarted';
+
+  let command = '';
+  if (selectedCluster) {
+    command = generateCmd({
+      namespace: 'teleport-agent',
+      clusterName: selectedCluster.name,
+      proxyAddr: ctx.storeUser.state.cluster.publicURL,
+      tokenId: joinToken.id,
+      clusterVersion: ctx.storeUser.state.cluster.authVersion,
+      resourceId: joinToken.internalResourceId,
+      isEnterprise: ctx.isEnterprise,
+      isCloud: ctx.isCloud,
+      automaticUpgradesEnabled: ctx.automaticUpgradesEnabled,
+      automaticUpgradesTargetVersion: ctx.automaticUpgradesTargetVersion,
+      joinLabels: [...selectedCluster.labels, ...selectedCluster.joinLabels],
+      disableAppDiscovery: !isAppDiscoveryEnabled,
+    });
+  }
+
+  return (
+    <Box maxWidth="1000px">
+      <Header>Enroll an EKS Cluster</Header>
+      {fetchClustersAttempt.status === 'failed' && !hasIamPermError && (
+        <Danger mt={3}>{fetchClustersAttempt.statusText}</Danger>
+      )}
+      <Text mt={4}>
+        Select the AWS Region you would like to see EKS clusters for:
+      </Text>
+      <AwsRegionSelector
+        onFetch={fetchClustersWithNewRegion}
+        onRefresh={refreshClustersList}
+        clear={clear}
+        disableSelector={fetchClustersAttempt.status === 'processing'}
+      />
+      {!hasIamPermError && tableData.currRegion && (
+        <>
+          <Box mb={2}>
+            <Toggle
+              isToggled={isAppDiscoveryEnabled}
+              onToggle={() => setAppDiscoveryEnabled(!isAppDiscoveryEnabled)}
+            >
+              <Box ml={2} mr={1}>
+                Enable Kubernetes App Discovery
+              </Box>
+              <ToolTipInfo>
+                Teleport's Kubernetes App Discovery will automatically identify
+                and enroll to Teleport HTTP applications running inside a
+                Kubernetes cluster.
+              </ToolTipInfo>
+            </Toggle>
+          </Box>
+          <ClustersList
+            items={tableData.items}
+            fetchStatus={tableData.fetchStatus}
+            selectedCluster={selectedCluster}
+            onSelectCluster={setSelectedCluster}
+            fetchNextPage={fetchNextPage}
+          />
+          <StyledBox mb={5} mt={5}>
+            <Text mb={2}>Automatically enroll selected EKS cluster</Text>
+            <ButtonSecondary
+              width="215px"
+              type="submit"
+              onClick={handleOnEnroll}
+              disabled={enrollmentNotAllowed}
+              mt={2}
+              mb={2}
+            >
+              Enroll EKS Cluster
+            </ButtonSecondary>
+            <Box>
+              <ButtonText
+                disabled={enrollmentNotAllowed}
+                onClick={() => {
+                  setIsManualHelmDialogShown(b => !b);
+                }}
+                pl={0}
+              >
+                Or click here to see instructions for manual enrollment
```

Comment:
What would you say about something like this instead?

<img width="708" alt="enroll-eks" src="https://github.com/gravitational/teleport/assets/27113/3d10de91-c040-499b-bcc6-0e1fbe2a5687">

I'm not a fan of saying "Click here" because the browser already has UI elements that prompt the user to click on them stuff – they are links and buttons. ;) In case of links on most websites, or `ButtonText` here, the problem is that there's no underline of any kind, hence "Click here" to help the user realize that they can click somewhere.

I also changed "Enroll EKS Cluster" to a primary button but I'm not sure if that's an action that we want to "promote" over manual setup.

```diff
diff --git a/web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx b/web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx
index 275cf3d71f..c9628b9163 100644
--- a/web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx
+++ b/web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx
@@ -17,7 +17,7 @@
  */
 
 import React, { useState } from 'react';
-import { Box, ButtonSecondary, ButtonText, Text, Toggle } from 'design';
+import { Box, ButtonSecondary, Text, Toggle, Flex } from 'design';
 import styled from 'styled-components';
 import { FetchStatus } from 'design/DataTable/types';
 import { Danger } from 'design/Alert';
@@ -328,27 +328,24 @@ export function EnrollEksCluster(props: AgentStepProps) {
           />
           <StyledBox mb={5} mt={5}>
             <Text mb={2}>Automatically enroll selected EKS cluster</Text>
-            <ButtonSecondary
-              width="215px"
-              type="submit"
-              onClick={handleOnEnroll}
-              disabled={enrollmentNotAllowed}
-              mt={2}
-              mb={2}
-            >
-              Enroll EKS Cluster
-            </ButtonSecondary>
-            <Box>
-              <ButtonText
+            <Flex alignItems="flex-start" gap={2}>
+              <ButtonSecondary
+                width="215px"
+                type="submit"
+                onClick={handleOnEnroll}
+                disabled={enrollmentNotAllowed}
+              >
+                Enroll EKS Cluster
+              </ButtonSecondary>
+              <ButtonSecondary
                 disabled={enrollmentNotAllowed}
                 onClick={() => {
                   setIsManualHelmDialogShown(b => !b);
                 }}
-                pl={0}
               >
-                Or click here to see instructions for manual enrollment
-              </ButtonText>
-            </Box>
+                View instructions for manual enrollment
+              </ButtonSecondary>
+            </Flex>
           </StyledBox>
         </>
       )}

```

Reply:
We actually don't want to give attention to the manual option, it's kind of last resort fallback, so kept very low key. For now I'll leave it as it is, but we can change it later. I'll ask about this on our next Discover sync meeting 👍